tests/drivers/spi/spi_slave_master_loopback/src/spi.c

```diff
@@ -0,0 +1,424 @@
+/*
+ * Copyright (c) 2017 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define SYS_LOG_LEVEL SYS_LOG_LEVEL_DEBUG
+#include <logging/sys_log.h>
+
+#include <zephyr.h>
+#include <misc/printk.h>
+#include <string.h>
+#include <stdio.h>
+
+#include <spi.h>
+
+#define SPI_MASTER_CS &spi_master_cs
+#define SPI_SLAVE_CS &spi_slave_cs
+
+#if defined(CONFIG_BOARD_NUCLEO_F091RC)
```

Comment:
Following the led sample app pattern I referred to in another comment, to make this easier to port to other boards, how about doing following something like this here instead:

```
#define SPI_MASTER_DRV_NAME "spi-loopback-master"
#define SPI_SLAVE_DRV_NAME "spi-loopback-slave"
```

Then in the nucleo f091rc .conf fragment:

```
CONFIG_SPI_1_NAME="spi-loopback-master"
CONFIG_SPI_2_NAME="spi-loopback-slave"
```

Then device_get_binding(SPI_xxx_DRV_NAME) can still be used to find the device, and additional boards be ported to this test without adding anything to the source code.

Similarly, an application-specific Kconfig file (specified with KCONFIG_ROOT; see the application development primer for details) could be used to configure the rest -- SPI_SLAVE, MIN_FREQ, etc. -- in a board-specific conf file. (Or perhaps DT would be better -- @nashif / @galak any opinion?)

Reply:
Good ideas aswell, thanks

Reply:
I did the DRV_NAME trick, I think the other stuff will need to wait until DT has been implemented for SPI